driverapi/driverapi.go

```diff
@@ -14,6 +14,9 @@ type Driver interface {
 	// Push driver specific config to the driver
 	Config(options map[string]interface{}) error
 
+	// Capabilities of the driver
+	Capabilities() Capability
```

Comment:
Agreed. This calls for a change in controller data-structure to maintain the capabilities for each driver being registered. That results in larger change on all the places `network.driver` was used. Hence took this simpler approach. But, since you feel this can be a potential and real problem, I will make the change of maintaining this state in controller and keep the code churn minimal.